src/libecalc/presentation/yaml/mappers/component_mapper.py

```diff
@@ -134,6 +139,16 @@ def from_yaml_to_dto(
                 except ValidationError as e:
                     raise DtoValidationError(data=data, validation_error=e) from e
 
+            elif component_type == ComponentType.PUMP_V2:
+                try:
+                    pump_yaml = YamlPump(**data)
+                    return pump_yaml.to_temporal_domain_models(
+                        references=self.__references, timesteps=timevector, fuel=fuel
+                    )
```

Comment:
Can't this be done in energy calculator instead? So we don't have to send timevector here?

Reply:
not sure, didnt check, but since we are moving away from energy calculator?